plugins/languages/debugprint.nix

```diff
@@ -5,33 +5,111 @@
   pkgs,
   ...
 }:
-with lib; let
-  cfg = config.plugins.debugprint;
-in {
-  options.plugins.debugprint =
-    helpers.neovim-plugin.extraOptionsOptions
-    // {
-      enable = mkEnableOption "debugprint.nvim";
-
-      package = helpers.mkPackageOption "debugprint.nvim" pkgs.vimPlugins.debugprint-nvim;
-
-      createCommands = helpers.defaultNullOpts.mkBool true ''
-        Creates default commands.
-      '';
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "debugprint";
+    originalName = "debugprint.nvim";
+    defaultPackage = pkgs.vimPlugins.debugprint-nvim;
 
-      createKeymaps = helpers.defaultNullOpts.mkBool true ''
-        Creates default keymappings.
-      '';
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-04-07: remove 2024-06-07
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = [
+      "moveToDebugline"
+      "displayCounter"
+      "displaySnippet"
+      "filetypes"
```

Comment:
We can't simply use a renamedToSettings, as the filetypes ooptions is an attrsOf option that uses `midVar` and `rightVar` as keys. We either need to  mark it removed, or do something more intelligent as compatibilty.

Reply:
Indeed, done.